<a href="https://colab.research.google.com/github/sangeun0/Learning-python/blob/main/list_practice.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 문제 1.
- 문자 10개를 저장할 수 있는 리스트를 만들고 10개의 문자를 입력받아 입력받은 문자를 이어서 출력하는 프로그램을 작성하시오.
- 입력 예: A B C D E F G H I J
- 출력 예: ABCDEFGHIJ

In [ ]:
word1=[] #리스트 설정
for word in range(10): #문자 10개 입력
  word= input( "enter the word: " )
  word1.append(word) #입력되는 문자 리스트로 추가
  s= "".join(word1) #출력값 변수로 변환 한줄로 출력
print(s)

enter the word: a
enter the word: b
enter the word: c
enter the word: d
enter the word: e
enter the word: f
enter the word: g
enter the word: h
enter the word: i
enter the word: j
abcdefghij


# 문제 2.
- 10개의 문자를 입력받아서 첫 번째 네 번째 일곱 번째 입력받은 문자를 차례로 출력하는 프로그램을 작성하시오.
- 입력 예: A B C D E F G H I J
- 출력 예: A D G

In [ ]:
word1=[] #리스트 설정
for word in range(10): #문자 10개 입력
  word= input( "enter the word: " )
  word1.append(word) #입력되는 문자 리스트로 추가
  
print(word1[0],word1[3],word1[6]) #첫번째, 네번째 ,일곱 번째 문자 출력

enter the word: d
enter the word: f
enter the word: g
enter the word: h
enter the word: j
enter the word: r
enter the word: c
enter the word: w
enter the word: r
enter the word: y
d h c


# 문제 3.
- 빈 리스트을 선언하고 정수를 차례로 입력받다가 0 이 입력되면 0 을 제외하고 그 때까지 입력된 정수를 가장 나중에 입력된 정수부터 차례대로 출력하는 프로그램을 작성하시오.
- 입력 예: 
    * 3 
    * 5
    * 10 
    * 55 
    * 0
- 출력 예: 55 10 5 3

In [ ]:
number1= []

while True:
  number= int(input( "enter the number: "))
  if number==0 :
    break
  if number>0:
    number1.append(number) #리스트에 값 추가

#뒤에서부터 슬라이싱
print(number1[::-1])

enter the number: 4
enter the number: 6
enter the number: 7
enter the number: 8
enter the number: 0
[8, 7, 6, 4]


# 문제 4.
- 1반부터 6반까지의 평균점수를 저장한 후 두 반의 반 번호를 입력받아 두 반 평균점수의 합을 출력하는 프로그램을 작성하시오.
- 반별 평균점수는 초기값으로 1반부터 차례로 85.6 79.5 83.1 80.0 78.2 75.0으로 초기화하고, 
- 출력은 소수 두 번째 자리에서 반올림하여 소수 첫째자리까지 한다.
- 입력 예: 1 3 (1반 3반을 의미)
- 출력 예: 168.7

In [ ]:
class_float = [60 ,85.6 ,79.5 ,83.1 ,80.0 ,78.2, 75.0] #리스트 소수설정, 1반부터 시작> 입력값과 반점수 일치할 수 있도록 0번리스트 임의로 지정

class1= int(input("enter the class1: "))
class2= int(input("enter the class2: "))
#반 입력받기
avg= (class_float[class1]+class_float[class2])/2 #입력받은 반번호를 class_float에서 index값으로 찾기
print(round(avg,1))

enter the class1: 2
enter the class2: 5
78.8


# 문제 5.
- 10개의 정수를 입력받아 그 중 가장 작은 수를 출력하는 프로그램을 작성하시오.
- 입력 예: 5 10 8 55 6 31 12 24 61 2
- 출력 예: 2

In [ ]:

for number in range(10):
  number= input("enter the number: ") #숫자 10개 입력

print(min(number)) #최솟값구하기

enter the number: 21
enter the number: 43
enter the number: 56
enter the number: 43
enter the number: 23
enter the number: 67
enter the number: 87
enter the number: 6
enter the number: 32
enter the number: 4
4


# 문제 6.
- 10개의 정수를 입력받아 100 미만의 수 중 가장 큰 수와 100 이상의 수 중 가장 작은 수를 출력하는 프로그램을 작성하시오.
- 입력 예: 88 123 659 15 443 1 99 313 105 48
- 출력 예: 99 105

In [ ]:
number1= []
for number in range(10):
  number= int(input("enter the number: ")) #숫자 10개 입력
  number1.append(number)
  
  if number<100 :
    maxnum= max(number)

  if number>=100 :
    minnum= min(number)

print(maxnum)
print(minnum) #최솟값구하기

enter the number: 32


TypeError: ignored

# 문제 7.
- 10개의 정수를 입력받아 리스트에 저장한 후 짝수 번째 입력된 값의 합과 홀수 번째 입력된 값의 평균을 출력하는 프로그램을 작성하시오.
- 평균은 반올림하여 소수첫째자리까지 출력한다.
- 입력 예: 95 100 88 65 76 89 58 93 77 99
- 출력 예: 
    * sum : 446
    * avg : 78.8

In [ ]:
number1 =[]


for number in range(10):
  number=int(input( "enter the number: " ))
  number1.append(number)

print(number1)
odd= (number1[0:10:2]) #홀수슬라이싱
even= (number1[1:10:2]) #짝수슬라이싱
print( even)
print( odd)

print("짝수번째 덧셈", sum(even))
print("홀수번째 평균", sum(odd)/len(odd))
 

enter the number: 1
enter the number: 2
enter the number: 3
enter the number: 4
enter the number: 5
enter the number: 6
enter the number: 7
enter the number: 8
enter the number: 9
enter the number: 1
[1, 2, 3, 4, 5, 6, 7, 8, 9, 1]
[2, 4, 6, 8, 1]
[1, 3, 5, 7, 9]
짝수 덧셈 21
홀수 평균 5.0


In [ ]:
()

# 문제 8.
- 10개의 정수를 입력받아 리스트에 저장한 후 내림차순으로 정렬하여 출력하시오.
- 입력 예: 95 100 88 65 76 89 58 93 77 99
- 출력 예: 100 99 95 93 89 88 77 76 65 58

In [ ]:
number1= []


for number in range(10):
  number= int(input( "enter the number: " ))
  number1.append(number)

number2 = sorted(number1, reverse=True) #리스트 내림차순 정리
print(number2)
  

enter the number: 3
enter the number: 2
enter the number: 1
enter the number: 5
enter the number: 6
enter the number: 9
enter the number: 7
enter the number: 8
enter the number: 34
enter the number: 76
[76, 34, 9, 8, 7, 6, 5, 3, 2, 1]


[1, 2, 3, 4, 12, 21, 34, 54]
